dts/bindings/video/ovti,ov5640.yaml

```diff
@@ -5,7 +5,7 @@ description: OV5640 CMOS video sensor
 
 compatible: "ovti,ov5640"
 
-include: i2c-device.yaml
+include: [i2c-device.yaml, video-interfaces.yaml]
```

Comment:
I think this will not have any effects as the properties defined in `video-interfaces.yaml` are for an**endpoint** node, not for the ov5640 node itself. The difficulty here (and the reason why I let https://github.com/zephyrproject-rtos/zephyr/pull/74415 always as draft) is that we cannot know the **endpoint** is actually a child or a grand child node of the video device node (e.g. ov5640) as **ports** node is optional, so that we don't know using 1 or 2 levels of **child-binding** inb the video-interfaces.yaml:

- child-binding:
     child-binding:

So, I think the solution here is that either we:
- Don't include` video-interfaces.yaml`, and just declare the properties in the devicetree. It will compile without any problem.
- Or add `compatible: "nxp,zephyr-endpoint"` in the endpoint node and in `video-interfaces.yaml`

@josuah Do you have a solution for this ?

Reply:
Dear, @ngphibang and @josuah,

If I change from `include: [i2c-device.yaml, video-interfaces.yaml] `to `include: i2c-device.yaml`, I got the compile error below:


`devicetree error: 'bus-type' appears in /soc/i2c@58001c00/ov5640@3c in /home/charlesdias/zephyrproject/zephyr/build/zephyr/zephyr.dts.pre, but is not declared in 'properties:' in /home/charlesdias/zephyrproject/zephyr/dts/bindings/video/ovti,ov5640.yaml`

Reply:
I think it's because you declared these properties on the ov5640 node. All the properties in `video-interfaces.yaml` should be declared in the **endpoint** subnode.